homeassistant/helpers/device_registry.py

```diff
@@ -257,6 +257,7 @@ class DeviceEntry:
     model: str | None = attr.ib(default=None)
     name_by_user: str | None = attr.ib(default=None)
     name: str | None = attr.ib(default=None)
+    primary_config_entry: str | None = attr.ib(default=None)
```

Comment:
Ok, I have changed my mind on this one. 

Initially, I was like: yo, let's just track 1. There should be only 1. 

But… Matter 🤔 

So I think that we should consider making this a list. With the first one being the absolute primary, and that is what is shown in the UI. And we should ensure that primary device entries coming in that belong to open standards or user defined standards (MQTT), are always appended to the list, while other entries can take the primary spot. 

Most devices will have a single entry, and we should always show the first, ie `["shelly"]`. Matter devices can have two, if they also have a native integration, ie `["lifx", "matter"]`. At this point, no one I know will ever have 3 …

Reply:
Same